ecosystem/sep-0036.md

```diff
@@ -0,0 +1,137 @@
+- `approval_server` is the url of an approval service that signs validated transactions.
+- `approval_criteria` is a human readable string that explains the issuer's requirements for approving transactions.
+
+### Example
+
+```toml
+[[CURRENCIES]]
+code="GOAT"
+issuer="GD5T6IPRNCKFOHQWT264YPKOZAWUMMZOLZBJ6BNQMUGPWGRLBK3U7ZNP"
+regulated=true
+approval_server="https://goat.io/tx_approve"
+approval_criteria="The goat approval server will ensure that transactions are compliant with NFO regulation"
+```
+
+## Approval Server
+
+The transaction approval server receives a signed transaction and checks for compliance and signs it on success.
+
+### Possible results
+- Success: Transaction was found compliant and signed by service.
+- Rejected: Transaction was rejected.
+
+### Request
+
+Transaction approval requests are executed by submitting an `HTTP POST` to `approval_server` with `Content-Type` being `application/x-www-form-urlencoded` or `application/json` and a single `transaction` parameter. `transaction` value is a base64 encoded transaction envelope XDR signed by the user. This is the transaction that will be tested for compliance and signed on success.
+
+### Responses
+
+HTTP responses have an `application/json` encoded body. All responses will contain, at least, a top level `status` parameter that indicates the type of the result.
+
+#### Success Response
+
+A Successful response will have a `200` HTTP status code and `success` as the `status` value. This response means that the transaction was found compliant and signed.
+
+Parameters:
+
+Name | Data Type | Description
+-----|-----------|------------
+status|string|`"success"`
+transaction|string|Transaction envelope XDR, base64 encoded. This transaction will have the original signature(s) from the request as well as the asset issuer's signature.
+message|string|A human readable string containing information to pass on to the user (optional).
+
+#### Rejected Response
+
+A Rejected response will have a `422` HTTP status code and `rejected` as the `status` value. It means that the transaction is not compliant and thus not signed by the server.
+
+The error message should explain why the transaction is not compliant and advice further actions for the client if possible. If there are further actions that the user must complete before this transaction can be approved, the issuer may provide a URL via `action_url` to faciliate the actions. If the issuer needs to asynchronously validate the transaction, the adviced wait time should be included in the error message if possible so the client would know when to retry.
+
+Parameters:
+
+Name | Data Type | Description
+-----|-----------|------------
+status|string|`"rejected"`
+error|string|A human readable string explaining why the transaction is not compliant.
+action_url|string|A URL that allows the user to complete the actions required to have the transaction approved (optional).
+
+## Transaction Composition
```

Comment:
I think this section should be much higher, since it explains how this actually works. Maybe after `Example`?